File: backend/src/bond_monitor/domain/trading/holdings.py
"""Trading holdings read-model."""

from __future__ import annotations

from dataclasses import dataclass
from datetime import date


@dataclass(frozen=True)
class HoldingView:
    """Позиция на счёте, обогащённая рыночными данными."""

    figi: str
    isin: str
    name: str
    lots: int
    quantity: int
    lot_size: int
    current_price_pct: float | None
    current_nkd_rub: float | None
    ytm: float | None
    maturity_date: date | None
    offer_date: date | None
    market_value_rub: float | None

File: backend/internal/domain/trading/holdings.go
package trading

import "time"

// HoldingView is a broker position enriched with market data.
type HoldingView struct {
	FIGI             string
	ISIN             string
	Name             string
	Lots             int
	Quantity         int
	LotSize          int
	CurrentPricePct  *float64
	CurrentNKDRub    *float64
	YTM              *float64
	MaturityDate     *time.Time
	OfferDate        *time.Time
	MarketValueRub   *float64
}
